crates/apub/src/api/list_posts.rs

```diff
@@ -87,6 +91,13 @@ pub async fn list_posts(
   };
   let page_back = data.page_back;
 
+  let community_id = match (data.community_id, data.multi_community_id) {
+    (Some(id), None) => Some(CommunityOrMulti::CommunityId(id)),
+    (None, Some(id)) => Some(CommunityOrMulti::MultiCommunityId(id)),
+    (Some(_), Some(_)) => return Err(LemmyErrorType::InvalidCommunityId.into()),
+    (None, None) => None,
+  };
```

Comment:
Any suggestions how the api and Postquery para should look like? At the moment it feels a bit awkward.

Reply:
I removed the CommunityOrMulti enum, but queries are already working fine and are better this way.